pkgs/development/python-modules/torch/mk-torch-compile-check.nix

```diff
@@ -1,38 +1,34 @@
-{
-  cudaPackages,
-  feature ? null,
-  lib,
-  libraries,
-  name ? if feature == null then "torch-compile-cpu" else "torch-compile-${feature}",
-  pythonPackages,
-  stdenv,
-}:
-let
-  deviceStr = if feature == null then "" else '', device="cuda"'';
-in
-(cudaPackages.writeGpuTestPython.override { python3Packages = pythonPackages; })
-  {
-    inherit name feature libraries;
-    makeWrapperArgs = [
-      "--suffix"
-      "PATH"
-      ":"
-      "${lib.getBin stdenv.cc}/bin"
-    ];
-  }
-  ''
-    import torch
+{ cudaPackages, feature ? null, lib, libraries, name ?
+  if feature == null then "torch-compile-cpu" else "torch-compile-${feature}"
+, pythonPackages, stdenv, }:
+let deviceStr = if feature == null then "" else '', device="${feature}"'';
+in (cudaPackages.writeGpuTestPython.override {
+  python3Packages = pythonPackages;
+}) {
+  inherit name feature libraries;
+  makeWrapperArgs = [ "--suffix" "PATH" ":" "${lib.getBin stdenv.cc}/bin" ];
+} (''
+  import torch
 
 
-    @torch.compile
-    def opt_foo2(x, y):
-        a = torch.sin(x)
-        b = torch.cos(y)
-        return a + b
+'' + lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
+  # torch.compile requires OpenMP which is not available on Darwin
```

Comment:
let's avoid putting comments in nix strings. they mess with the drv hash

Reply:
Fixed this in latest push